app/code/Magento/Eav/Model/Api/SearchCriteria/CollectionProcessor/FilterProcessor.php

```diff
@@ -64,20 +64,55 @@ private function addFilterGroupToCollection(
         AbstractDb $collection
     ) {
         $fields = [];
+        $customFilters = [];
         foreach ($filterGroup->getFilters() as $filter) {
             $isApplied = false;
+            $applyLater = false;
             $customFilter = $this->getCustomFilterForField($filter->getField());
             if ($customFilter) {
-                $isApplied = $customFilter->apply($filter, $collection);
+                if ($filter->getConditionType() == 'eq') {
```

Comment:
Please use strict comparison whenever it's possible.
```suggestion
                if ($filter->getConditionType() === 'eq') {
```